src/pages/account/settings/components/GeographicView.tsx

```diff
@@ -0,0 +1,175 @@
+import React, { Component } from 'react';
+import { Select, Spin } from 'antd';
+
+import { Dispatch } from 'redux';
+import { connect } from 'dva';
+import { GeographicItemType } from '../data.d';
+import styles from './GeographicView.less';
+
+const { Option } = Select;
+
+interface SelectItem {
+  label: string;
+  key: string;
+}
+const nullSelectItem: SelectItem = {
+  label: '',
+  key: '',
+};
+
+interface GeographicViewProps {
+  dispatch?: Dispatch<any>;
+  province?: GeographicItemType[];
+  city?: GeographicItemType[];
+  value?: {
+    province: SelectItem;
+    city: SelectItem;
+  };
+  loading?: boolean;
+  onChange?: (value: { province: SelectItem; city: SelectItem }) => void;
+}
+
+@connect(
+  ({
+    accountAndsettings,
+    loading,
+  }: {
+    accountAndsettings: {
+      province: GeographicItemType[];
+      city: GeographicItemType[];
+    };
+    loading: any;
+  }) => {
+    const { province, city } = accountAndsettings;
+    return {
+      province,
+      city,
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The key 'city' is not sorted alphabetically](https://app.codacy.com/app/afc163/ant-design-pro/pullRequest?prid=4583897)